### PR TITLE
ofRectangle::intersects(const ofPoint& p0, const ofPoint& p1) fails in one case. Fixes #4871

### LINKED ISSUE #4871: ofRectangle::intersects(const ofPoint& p0, const ofPoint& p1) fails in one case.
This should return true:

`ofRectangle(0, 0, 100, 100).intersects(ofPoint(0, 50), ofPoint(50, 0))`  It does not.

Thanks @anikahhh!

### PR DESCRIPTION
**Pull request by bakercp**
See #4871 for discussion.

### CODE REVIEW

File: libs/openFrameworks/types/ofRectangle.cpp [hunk 1]
```diff
@@ -407,19 +407,27 @@ bool ofRectangle::intersects(const ofRectangle& rect) const {
...
+	float left = standardized.getLeft();
+
+    return standardized.inside(p0) || // check end inside
+           standardized.inside(p1) || // check end inside
            ofLineSegmentIntersection(p0, p1, topLeft,     topRight,    p) || // cross top
            ofLineSegmentIntersection(p0, p1, topRight,    bottomRight, p) || // cross right
            ofLineSegmentIntersection(p0, p1, bottomRight, bottomLeft,  p) || // cross bottom
-           ofLineSegmentIntersection(p0, p1, bottomLeft,  topLeft,     p);   // cross left
+           ofLineSegmentIntersection(p0, p1, bottomLeft,  topLeft,     p) || // cross left
+		   (left == p0.x && top == p1.y) || // Special case 1.
```
Comment (arturoc) [inline libs/openFrameworks/types/ofRectangle.cpp]: comparing floats with == is not a good idea, this should be using some kind of epsilon or something like that

**Comment by arturoc** [comment]
can you change the title of this issue to something more descriptitve? i

File: libs/openFrameworks/types/ofRectangle.cpp (hunk 1 above)
Comment (bakercp) [inline libs/openFrameworks/types/ofRectangle.cpp]: Yes, was following the style in other places from the file.  I'll update all.

**Comment by bakercp** [comment]
@arturoc Since this issue is not limited to this one change in ofRectangle, I'd love to fix it by incorporating the function proposed here: #4888.

Let me know what you think.

**Comment by bakercp** [comment]
This now includes updates from #4958.

**Comment by arturoc** [comment]
i'll check this when i have a moment, i mostly need to merge it in the glm branch before merging it into master to be sure that i merge this individually since ofRectangle has changed quite a bit with glm

**Comment by bakercp** [comment]
@arturoc This should be ready to merge after CI checks.

**Comment by bakercp** [comment]
@arturoc This should be ready to merge after CI checks.

**Comment by bakercp** [comment]
@arturoc - This should be ready to go.

**Pull request closed without merging**